encoding/proto/proto.go

```diff
@@ -38,73 +35,14 @@ func init() {
 // codec is a Codec implementation with protobuf. It is the default codec for gRPC.
 type codec struct{}
 
-type cachedProtoBuffer struct {
-	lastMarshaledSize uint32
-	proto.Buffer
-}
-
-func capToMaxInt32(val int) uint32 {
-	if val > math.MaxInt32 {
-		return uint32(math.MaxInt32)
-	}
-	return uint32(val)
-}
-
-func marshal(v interface{}, cb *cachedProtoBuffer) ([]byte, error) {
-	protoMsg := v.(proto.Message)
-	newSlice := make([]byte, 0, cb.lastMarshaledSize)
-
-	cb.SetBuf(newSlice)
-	cb.Reset()
-	if err := cb.Marshal(protoMsg); err != nil {
-		return nil, err
-	}
-	out := cb.Bytes()
-	cb.lastMarshaledSize = capToMaxInt32(len(out))
-	return out, nil
-}
-
 func (codec) Marshal(v interface{}) ([]byte, error) {
-	if pm, ok := v.(proto.Marshaler); ok {
```

Comment:
This is breaking existing code generate with `gogoproto.customtype = "MyType"`, where a user can write custom marshaling/unmarshaling to have more control.

https://github.com/gogo/protobuf/blob/master/custom_types.md#custom-type-method-signatures

Reply:
`gogoproto` is not officially supported by gRPC-Go.

If you want a codec that suits your needs better, it's easy to install your own.  Just copy/paste this code into another package, update the parts that are important, and `import` it in your `main` package after any `grpc` imports.  (If you don't control `main`, you can give it a new name and use the `Codec` `Call`-/`Dial`-/`Server`- `Option`s.)

Reply:
Ok thanks, I will give it a try. I have also created an issue about this. Please feel free to close it #4192 I will also comment the solution there after testing it.